src/SonarScanner.Shim/PropertiesFileGenerator.cs

```diff
@@ -158,28 +158,28 @@ public bool TryWriteProperties(PropertiesWriter writer, out IEnumerable<ProjectD
         ///     This method has some side effects.
         /// </remarks>
         /// <returns>The list of files to attach to the root module.</returns>
-        private ICollection<string> PutFilesToRightModuleOrRoot(IEnumerable<ProjectData> projects, string rootProjectBaseDir)
+        private ICollection<FileInfo> PutFilesToRightModuleOrRoot(IEnumerable<ProjectData> projects, DirectoryInfo baseDirectory)
         {
             var fileWithProjects = projects
                 .SelectMany(p => p.ReferencedFiles.Select(f => new { Project = p, File = f }))
-                .GroupBy(group => group.File, StringComparer.OrdinalIgnoreCase)
+                .GroupBy(group => group.File, new FileInfoEqualityComparer())
                 .ToDictionary(group => group.Key, group => group.Select(x => x.Project).ToList());
 
-            var rootModuleFiles = new List<string>();
+            var rootModuleFiles = new HashSet<FileInfo>(new FileInfoEqualityComparer());
 
             foreach (var group in fileWithProjects)
             {
                 var file = group.Key;
 
-                if (!File.Exists(file))
+                if (!file.Exists)
                 {
                     logger.LogWarning(Resources.WARN_FileDoesNotExist, file);
                     logger.LogDebug(Resources.DEBUG_FileReferencedByProjects, string.Join("', '",
                         group.Value.Select(x => x.Project.FullPath)));
                     continue;
                 }
 
-                if (!PathHelper.IsInFolder(file, rootProjectBaseDir)) // File is outside of the SonarQube root module
+                if (!PathHelper.IsInDirectory(file, baseDirectory)) // File is outside of the SonarQube root module
                 {
                     logger.LogWarning(Resources.WARN_FileIsOutsideProjectDirectory, file);
```

Comment:
Unrelated: might be worth dumping the base directory in the warning too to help when answering support questions.

Reply:
Actually it is already done as debug right after.

Reply:
Whereabouts? The LogDebug in line 185 doesn't include the _baseDirectory_.

I was actually thinking it would be better to have the data in the warning message anyway - that way it's less likely that we'll have to ask the customer to run the build again this time specifying _/d:sonar.debug_ etc.

Reply:
You are right it is not specified. I have updated the warning to include the base dir but kept the debug listing projects referencing this file.